File: archive/j/java/Factorial.java
public class Factorial {
    public static long fact(long n) {
        if (n <= 0)
            return 1;
        return n * fact(n - 1);
    }

    public static void main(String[] args) {
        try {
            long n = Long.parseLong(args[0]);
            if (n > 59) {
                System.out.println(String.format("%1$s! is out of the reasonable bounds for calculation.", n));
                System.exit(1);
            } else if (n < 0) {
                System.out.println("Usage: please input a non-negative integer");
                System.exit(1);
            }
            long result = fact(n);
            System.out.println(result);
        } catch (Exception e) {
            System.out.println("Usage: please input a non-negative integer");
            System.exit(1);
        }
    }
}

File: archive/j/julia/factorial.jl
#!/usr/bin/julia

function err() 
   println("Usage: please input a non-negative integer")
end

function fac(n)
    if n < 0
        err()
        exit()
    end
        
    out = 1
    for i = 1:n
        out = out * i
    end
    return out
end

try
    println(fac(parse(Int, ARGS[1])))
catch e
    err()
end
